packages/core/src/transactions/utils/simplePath.ts

```diff
@@ -0,0 +1,192 @@
+import { AbiItem, isAddress } from 'web3-utils'
+
+import TransactionPath from '../TransactionPath'
+import TransactionRequest from '../TransactionRequest'
+import { createDirectTransactionForApp } from './transactions'
+import App from '../../wrappers/App'
+import Organization from '../../wrappers/Organization'
+import { Permission } from 'plumbery-connector-thegraph/src/graphql/types'
+
+export interface AragonArtifactFunction {
+  roles: string[]
+  sig: string
+  /**
+   * This field might not be able if the contract does not use
+   * conventional solidity syntax and Aragon naming standards
+   * null if there in no notice
+   */
+  notice: string | null
+  /**
+   * The function's ABI element is included for convenience of the client
+   * null if ABI is not found for this signature
+   */
+  abi: AbiItem | null
+}
+
+export const ANY_ENTITY = '0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF'
+
+export function addressesEqual(first: string, second: string): boolean {
+  first = first && first.toLowerCase()
+  second = second && second.toLowerCase()
+  return first === second
+}
+
+// "Safer" version of [].includes() for addresses
+export function includesAddress(arr: any[], address: string): boolean {
+  return arr.some((a) => addressesEqual(a, address))
+}
+
+// TODO:
+// export function getAppsFromPath(transactions: TransactionRequest[]): App[] {
+//   return
+// }
+
+/**
+ * Calculate the transaction path for a transaction to `destination`
+ * that invokes `methodSignature` with `params`.
+ *
+ * @param  {string} [finalForwarder] Address of the final forwarder that can perfom the action.
+ *                  Needed for actions that aren't in the ACL but whose execution depends on other factors
+ * @return {Promise<TransactionPath>} An array of Ethereum transactions that describe each step in the path
+ */
+export async function calculateTransactionPath(
+  sender: string,
+  destination: string,
+  methodSignature: string,
+  params: string[],
+  org: Organization,
+  path?: string[],
+  as?: string,
+  finalForwarder?: string
+): Promise<TransactionPath | null> {
+  // Get the destination app
+  const app = await org.app(destination)
+  if (!app) {
+    throw new Error(
+      `Transaction path destination (${destination}) is not an installed app`
+    )
```

Comment:
On the errors: should we have specialized a module of specialized error types + names for the entire library, so that users can easily check them, either by name or by importing them + `typeof`?

Reply:
Yeah, there are not that many things that could go wrong and better have as clear errors as we can